src/main/java/net/dv8tion/jda/core/requests/Route.java

```diff
@@ -393,12 +369,36 @@ public String toString()
         private final Route baseRoute;
         private final String ratelimitRoute;
         private final String compiledRoute;
+        private final boolean hasQueryParams; 
 
-        private CompiledRoute(Route baseRoute, String ratelimitRoute, String compiledRoute)
+        private CompiledRoute(Route baseRoute, String ratelimitRoute, String compiledRoute, boolean hasQueryParams)
         {
             this.baseRoute = baseRoute;
             this.ratelimitRoute = ratelimitRoute;
             this.compiledRoute = compiledRoute;
+            this.hasQueryParams = hasQueryParams;
+        }
+
+        private CompiledRoute(Route baseRoute, String ratelimitRoute, String compiledRoute)
+        {
+            this(baseRoute, ratelimitRoute, compiledRoute, false);
+        }
+
+        public CompiledRoute withQueryParams(String name, String value)
+        {
+            return new CompiledRoute(baseRoute, ratelimitRoute, compiledRoute + (hasQueryParams ? '&' : '?') + name + '=' + value, true);
+        }
+
+        public CompiledRoute withQueryParams(String... params)
+        {
+            Checks.check(params.length % 2 == 0, "param length has to be even");
+
```

Comment:
Change message to 'params length must be a multiple of 2`.
Basically the same, but more clear.